docs/CONTRIBUTING.md

```diff
@@ -24,17 +340,64 @@ Alternative methods if you don't like Github (also fully supported):
  1. Attach them to <https://bugzilla.gnome.org/>
 
 It is likely however once a patch is ready to apply a maintainer
-will push it to a github PR, and merge via Homu.
+will push it to a github PR, and merge via Homu. -->
 
-Commit message style
---------------------
+## [Commit message style](#commit-message-style)
 
 Please look at `git log` and match the commit log style, which is very
 similar to the
 [Linux kernel](https://git.kernel.org/cgit/linux/kernel/git/torvalds/linux.git).
 
 You may use `Signed-off-by`, but we're not requiring it.
 
+**TODO**: Add rules about commit message and link the source, Editing committed message may be unnecessary...
+
+**General Commit Message Guidelines**:  
+
+1. Title
+    - Specify the context or category of the changes .e.g `lib` for library changes, `docs` for document changes, `bin/<command-name>` for command changes, etc.
+    - Begin the title with the first letter of the first word capitalized.
+    - Aim for less than 50 characters, otherwise 72 characters max.
+    - Do not end the title with a period.
+    - Use an [imperative tone](https://en.wikipedia.org/wiki/Imperative_mood).
+2. Body
+    - Separate the body with a blank line after the title.
+    - Begin a paragraph with the first letter of the first word capitalized.
+    - Each paragraph should have be within 72 characters.
+    - Content should be about what was changed and why this change was made.
+    - Closes: #issue-number. If the commit resolve an issues, otherwise it is not needed.
+
+
+Commit Message example:
+
+```bash
+<context>: Less than 50 Characters for subject title
+
+A paragraph of the body should be within 72 characters.
+
+This paragraph is also less than 72 characters.
+
+Closes: #issue-number
+```
+
+**Editting a Committed Message:**
+
+To edit the message from the most recent commit run `git commit --amend`. To change older commits on the branch use `git rebase -i`. For a successful rebase have the branch track `upstream master`. Once the changes have been made and saved, run `git push --force origin <branch-name>`.
+
+## [Returning workflow](#returning-workflow)
+
+When returning to make a new patch, do the following to update your fork with the latest changes to master:
+
+```bash
+git fetch upstream
+git merge upstream/master
```

Comment:
I'd recommend `git rebase upstream/master` instead.

Reply:
Tried `git rebase upstream/master` and it works better than doing fetch/merge - changing to rebase.